manifests/modules/automation/gitops/flux/.workshop/terraform/addon.tf

```diff
@@ -62,5 +62,496 @@ output "environment" {
   value = <<EOF
 export GITOPS_IAM_SSH_KEY_ID=${aws_iam_user_ssh_key.gitops.id}
 export GITOPS_IAM_SSH_USER=${aws_iam_user.gitops.unique_id}
+export IMAGE_URI_UI=${aws_ecr_repository.ecr_ui.repository_url}
 EOF
-}
\ No newline at end of file
+}
+
+## CI to Flux
+
+data "aws_vpc" "selected" {
+  tags = {
+    created-by = "eks-workshop-v2"
+    env        = local.addon_context.eks_cluster_id
+  }
+}
+
+data "aws_subnets" "private" {
+  tags = {
+    created-by = "eks-workshop-v2"
+    env        = local.addon_context.eks_cluster_id
+  }
+
+  filter {
+    name   = "tag:Name"
+    values = ["*Private*"]
+  }
+}
+
+data "aws_security_group" "default" {
+  vpc_id = data.aws_vpc.selected.id
+
+  filter {
+    name   = "group-name"
+    values = ["default"]
+  }
+}
+
+output "gitops_ecr_url_ui" {
+  value       = aws_ecr_repository.ecr_ui.repository_url
+  description = "ECR repository Url for UI module"
+}
+
+resource "aws_codecommit_repository" "codecommit_retail_store_sample" {
+  repository_name = "${local.addon_context.eks_cluster_id}-retail-store-sample"
+  description     = "CodeCommit repository for retail-store-sample"
+}
+
+data "aws_iam_policy_document" "ci_access" {
+  statement {
+    sid = ""
+    actions = [
+      "codecommit:GitPull",
+      "codecommit:GitPush"
+    ]
+    effect = "Allow"
+    resources = [
+      aws_codecommit_repository.codecommit_retail_store_sample.arn
+    ]
+  }
+}
+
+resource "aws_iam_policy" "ci_access" {
+  name   = "${local.addon_context.eks_cluster_id}-ci-access"
+  path   = "/"
+  policy = data.aws_iam_policy_document.ci_access.json
+}
+
+resource "aws_iam_user_policy_attachment" "ci_access" {
+  user       = aws_iam_user.gitops.name
+  policy_arn = aws_iam_policy.ci_access.arn
+}
+
+resource "aws_ecr_repository" "ecr_ui" {
+  name                 = "retail-store-sample-ui"
```

Comment:
Can you add a random suffix to this? Currently if this lab fails to clean up for some reason it will block any subsequent test runs without manual cleanup.

Reply:
I added changes from @csantanapr and random suffix for the ecr repository. I also enabled tests for all of the steps. It takes time to pass the tests and the last step need larger timeout, but we don't have skipped tests anymore.